inspect and list command updated (#3000)
conan-io/docs: reference/commands/inspect.rst

                            -vvv or -vtrace
      --logger              Show the output with log format, with time, type and
                            message.

Inspect default and json output

> -------------------------------

Reviewer: I think the title is not necessary here.

                            -vvv or -vtrace
      --logger              Show the output with log format, with time, type and
                            message.

> Inspect default and json output

Reviewer 2: Please tag ``conan inspect`` as preview, it was always a bit problematic command

-------------------------------

> This ``conan inspect`` default output shows the following information format.

Reviewer 2: It is worth adding an "important" note that says that ``conan inspect`` doesn't really evaluate any methods or apply any conditional logic. It list class attributes only.

.. code-block:: text

    $ conan inspect .
    default_options:
        shared: False
        fPIC: True
        neon: True
        msa: True
        sse: True
        vsx: True
        api_prefix:
    description: libpng is the official PNG file format reference library.
    generators: []
    homepage: http://www.libpng.org
    license: libpng-2.0
    name: libpng
    no_copy_source: False
    options:
        shared: [True, False]
        fPIC: [True, False]
        neon: [True, 'check', False]
        msa: [True, False]
        sse: [True, False]
        vsx: [True, False]
        api_prefix: ['ANY']
    revision_mode: hash
    settings: ('os', 'arch', 'compiler', 'build_type')
    topics: ('png', 'graphics', 'image')
    url: https://github.com/conan-io/conan-center-index

> The ``conan inspect ... --format=json`` will return a json output in ``stdout`` with the same information.

Reviewer: ```suggestion
The :command:`conan inspect ... --format=json` returns a JSON output format in ``stdout`` (can be redirected to file) with the following structure::
```

                            -vvv or -vtrace
      --logger              Show the output with log format, with time, type and
                            message.

.. note::

>     ``conan inspect`` doesn't really evaluate any methods or apply any conditional logic. It list class attributes only.

Reviewer 3: ```suggestion
    ``conan inspect`` doesn't really evaluate any methods or apply any conditional logic. It lists class attributes only.
```

The :command:`conan inspect` command shows the public attributes of any recipe (`conanfile.py`) as follows:

.. code-block:: text

    $ conan inspect .
    default_options:
        shared: False
        fPIC: True
        neon: True
        msa: True
        sse: True
        vsx: True
        api_prefix:
    description: libpng is the official PNG file format reference library.
    generators: []
    homepage: http://www.libpng.org
    license: libpng-2.0
    name: libpng
    no_copy_source: False
    options:
        shared: [True, False]
        fPIC: [True, False]
        neon: [True, 'check', False]
        msa: [True, False]
        sse: [True, False]
        vsx: [True, False]
        api_prefix: ['ANY']
    revision_mode: hash
    settings: ('os', 'arch', 'compiler', 'build_type')
    topics: ('png', 'graphics', 'image')
    url: https://github.com/conan-io/conan-center-index

> The :command:`conan inspect ... --format=json` returns a JSON output format in ``stdout`` (can be redirected to file) with the following structure::

Reviewer 3: ```suggestion
The :command:`conan inspect ... --format=json` returns a JSON output format in ``stdout`` (which can be redirected to a file) with the following structure::
```
